### DISCUSSION TITLE
How to make an enumeration in beamer with python and jupyter notebook (quarto)?

### DISCUSSION THREAD

**Discussion by Mgonrub** [quarto-dev/quarto-cli#11472] [Q&A]
### Description

Hello, I need help with the code shown in the image, which is what I want. But I have not been able to achieve it. Attached is the jupyter notebook file and the pdf of the result with the beamer.
I also tried another option adding ```{latex} and it didn't work either.
Any ideas please.
Thanks

[sample1a.pdf](https://github.com/user-attachments/files/17804764/sample1a.pdf)
[BBB.zip](https://github.com/user-attachments/files/17804771/BBB.zip)
![code](https://github.com/user-attachments/assets/ae5f144d-4d7a-4635-b359-fb62a7942998)

another option:

but the result is different. They are squares and not spheres:
![code2](https://github.com/user-attachments/assets/0f0ea015-7043-42b5-b28c-e6e74066982d)

`## Dats: U eje apliSTA.
```{=latex}
\begin{enumerate}[<+-| alert@+>]
\item (CHEERS) + MLE.
\item Estud JFK.
\item BRE 911.
\item LE Front (Embase + Medline in Ovid, Scielo).
\end{enumerate}`

**Comment by mcanouil** [comment]
Your question is mostly a LaTeX question and since you did not share a plain text markdown document, there is little we can do.

I suggest you read the guides for `beamer`:
- https://quarto.org/docs/presentations/beamer.html
- https://quarto.org/docs/reference/formats/presentations/beamer.html

And markdown basics guide, especially the section about using raw content: https://quarto.org/docs/authoring/markdown-basics.html#raw-content

For advanced debugging/styling, you can use `keep-tex: true` in your YAML frontmatter to take a look at the intermediate $\LaTeX$ document

For a more targeted help, please provide a small fully reproducible Quarto document.
You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by Mgonrub** [reply]
Hi,
I have left the jupyter file (sample1a.ipynb) inside the BBB.zip file.
[BBB.zip](https://github.com/user-attachments/files/17819218/BBB.zip)

![bbb](https://github.com/user-attachments/assets/268685f8-6b1b-4938-9206-a7c15bac41be)

I have converted sample1a.ipynb to quarto (qmd) and it is attached as well (sample1a.zip).
[sample1a.zip](https://github.com/user-attachments/files/17819205/sample1a.zip)

Thanks.

**Reply by mcanouil** [reply]
GitHub supports flat files. You don't need to attach them as ZIP.
Use code block and paste it as I showed in my previous post.
Pleas ensure, it's fully reproducible (it does not use external files for instance).
Note also that Jupyter Notebook is JSON, so you can also share it as plain text.

**Reply by mcanouil** [reply]
Please do read the links I shared. I'm pretty confident you are not following the documentation, especially the raw content part.
You should have all the needed information to solve your issue.

Edit: if you don't like the default LaTeX settings and in addition to the previous links see https://quarto.org/docs/journals/templates.html#latex-partials

**Reply by Mgonrub** [reply]
Hi guys.
I'm not an expert, that's why I'm asking. I'm just trying to learn. And Quarto for me is just an element that I want to see if it's useful to implement, since I'm also learning python. Thanks for your answers.

I've found a basic solution. It's not the best. But I'm closing the topic.
the result:
![code3](https://github.com/user-attachments/assets/9cb139a2-1e49-4989-9f36-ea1993eb6ea0)

For friends who want to learn, I'm sharing my non-expert solution (in jupyter notebook): In the cell at the beginning (Raw NBConvert), I refer to the "a YAML block", add "innertheme: rounded" that is, as the output will be beamer, provide the beamer (pdf) output options. In this case the format is beamer:



````qmd
---
title: Recur and Cost-Eff.
subtitle: GPC of 2024 ... GRADEpro.
description: ' Pandas, Matplotlib, Numpy y Seaborn.'
author:
  name: Marc
  url: www/

format:
  beamer:
    aspectratio: 32
    navigation: horizontal
    innertheme: rounded

jupyter: python3
---



Then in a jupyter notebook cell as Markdown:


`
```{=latex}

\begin{enumerate}[<+-| alert@+>]
    \item First argument
    \item Second argument
    \item Third argument
    \item Fourth argument
\end{enumerate}
```

`
With or without  ```{=latex} ```

**Reply by mcanouil** [reply]
Thanks for sharing the use of the `innertheme` option!

There are no LaTeX "expert" here. That's the thing.
Here you've decided to use Raw LaTeX instead of using markdown which is the base language of Quarto.
This means you know what you are doing and accept to be in the advanced usage area.
Most if not all the options described in the links I shared are Pandoc template option which are LaTeX options. Again, it means LaTeX beamer knowledge is required.

A simple usage here for incremental list as documented would lead to:

````qmd
---
title: Recur and Cost-Eff.
subtitle: GPC of 2024 ... GRADEpro.
description: Pandas, Matplotlib, Numpy y Seaborn.
author: Marc
format:
  beamer:
    aspectratio: 32
    navigation: horizontal
    innertheme: rounded
engine: python3
---

## Slide

Then in a jupyter notebook cell as Markdown:

::: {.incremental}
1. First argument
2. Second argument
3. Third argument
4. Fourth argument
:::
````

This feature is documented in Quarto guide:
- https://quarto.org/docs/presentations/beamer.html#incremental-lists

But actually comes straight from Pandoc:
- https://pandoc.org/chunkedhtml-demo/10.2-incremental-lists.html


The equivalent notebook (save the following JSON as `index.ipynb`):

```json
{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "---\n",
        "title: Recur and Cost-Eff.\n",
        "subtitle: GPC of 2024 ... GRADEpro.\n",
        "description: Pandas, Matplotlib, Numpy y Seaborn.\n",
        "author: Marc\n",
        "format:\n",
        "  beamer:\n",
        "    aspectratio: 32\n",
        "    navigation: horizontal\n",
        "    innertheme: rounded\n",
        "engine: python3\n",
        "---\n",
        "\n",
        "## Slide\n",
        "\n",
        "Then in a jupyter notebook cell as Markdown:\n",
        "\n",
        "::: {.incremental}\n",
        "1. First argument\n",
        "2. Second argument\n",
        "3. Third argument\n",
        "4. Fourth argument\n",
        ":::"
      ]
    }
  ],
  "metadata": {
    "kernelspec": {
      "name": "python3",
      "language": "python",
      "display_name": "Python 3 (ipykernel)",
      "path": "python3"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 4
}
```

**Reply by mcanouil** [reply]
FYI, `innertheme: rounded` is not the default in LaTeX and here where it is used in the template when defined:

- https://github.com/quarto-dev/quarto-cli/blob/bf93178346e68f721714ef4c7f792943b4d80f6e/src/resources/formats/pdf/pandoc/template.tex#L111-L113
